impress-org/givewp
# Hot Fix: multi-form goal block breaks from filter/form settings

Resolves #7068 and #6228 and #7070

## Description



The main thing that this PR fixes is the _"This block has encountered an error, and can not be previewed"_ error message displayed on the multi-form goal block when we try to filter the forms that should be used in the progress bar of the block.

The problem was happening because Gutenberg provides its own versions of React and ReactDOM. So, we have to tell third-party libraries about those versions by mapping the package names to the global variables available in the block editor.

Reference: https://github.com/JedWatson/react-select/issues/4893#issuecomment-1007551596

This PR also fixes a problem related to the form titles in the multi-select dropdown, it was displaying `&#8211;` instead of `-` in the form titles that contain this character.

This PR also fixes a bug that prevents a few shortcodes from respecting their own "ids" attribute, like for example these ones:

`[give_multi_form_goal ids="931"]`

`[give_form_grid ids="931"]`

`[give_totals ids="931" message="Thanks to you, we’ve raised {total} across all our congregations combined! Let's keep it up!" progress_bar="false"]`

## Affects



Multi-Form Goal Block > Progress Bar > Filters

## Visuals



![title-raw](https://github.com/impress-org/givewp/assets/16308864/a6bf5652-b3eb-4ae6-9d68-025f5f43fd00)

## Testing Instructions



1. Create a new page;
2. Add the multi-form goal block;
3. Click on the progress bar;
4. On the right side, click on Filters and Filter by form;
5. No more errors should be displayed.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205799879062080
  - https://app.asana.com/0/0/1205749635036779
